src/NCRings.jl

```diff
@@ -155,6 +155,13 @@ julia> is_unit(ZZ(-1)), is_unit(ZZ(4))
 """
 function is_unit end
 
+# Default implementation: covers all integral domains
+function is_nilpotent(a::T) where {T <: NCRingElem}
```

Comment:
```suggestion
function is_nilpotent(a::T) where {T <: NCRingElement}
```

Reply:
Changing `NCRingElem` to `NCRingElement` would appear to allow floating-point arguments; do we wish to allow that?   It would make the specializations for `Integer` and `Rational` redundant (as Max queried earlier).  My preference would be to keep the _status quo_.  Presumably, if we do make this change, then the analogous change should also be made to `is_unit` - right?

Reply:
BTW `is_unit` does have a method for `AbstractFloat` -- is that intentional?

Reply:
Change made, and pushed.